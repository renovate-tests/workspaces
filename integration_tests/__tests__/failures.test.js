/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @flow
 */

const path = require('path');
const skipOnWindows = require('../../scripts/skip_on_windows');
const {extractSummary} = require('../utils');
const runJest = require('../runJest');

const dir = path.resolve(__dirname, '../failures');

const normalizeDots = text => text.replace(/\.{1,}$/gm, '.');

skipOnWindows.suite();

const cleanupStackTrace = stderr => {
  const STACK_REGEXP = /^.*at.*(setup-jest-globals|extractExpectedAssertionsErrors).*\n/gm;
  if (!STACK_REGEXP.test(stderr)) {
    throw new Error(
      `
      This function is used to remove inconsistent stack traces between
      jest-jasmine2 and jest-circus. If you see this error, that means the
      stack traces are no longer inconsistent and this function can be
      safely removed.

      output:
      ${stderr}
    `,
    );
  }

  return (
    stderr
      .replace(STACK_REGEXP, '')
      // Also remove trailing whitespace.
      .replace(/\s+$/gm, '')
  );
};

test('not throwing Error objects', () => {
  let stderr;
  stderr = runJest(dir, ['throw_number.test.js']).stderr;
  expect(extractSummary(stderr).rest).toMatchSnapshot();
  stderr = runJest(dir, ['throw_string.test.js']).stderr;
  expect(extractSummary(stderr).rest).toMatchSnapshot();
  stderr = runJest(dir, ['throw_object.test.js']).stderr;
  expect(extractSummary(stderr).rest).toMatchSnapshot();
  stderr = runJest(dir, ['assertion_count.test.js']).stderr;
  expect(extractSummary(cleanupStackTrace(stderr)).rest).toMatchSnapshot();
});

test('works with node assert', () => {
  const {stderr} = runJest(dir, ['node_assertion_error.test.js']);
  expect(normalizeDots(extractSummary(stderr).rest)).toMatchSnapshot();
});
