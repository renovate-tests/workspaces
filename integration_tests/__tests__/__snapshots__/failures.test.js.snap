// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not throwing Error objects 1`] = `
" FAIL  __tests__/throw_number.test.js
  ● Test suite failed to run

    Expected an Error, but \\"1\\" was thrown

"
`;

exports[`not throwing Error objects 2`] = `
" FAIL  __tests__/throw_string.test.js
  ● Test suite failed to run

    Error
      banana

"
`;

exports[`not throwing Error objects 3`] = `
" FAIL  __tests__/throw_object.test.js
  ● Test suite failed to run

    Error: No message was provided

"
`;

exports[`not throwing Error objects 4`] = `
" FAIL  __tests__/assertion_count.test.js
  ● .assertions() › throws
    expect(received).toBeTruthy()
    Expected value to be truthy, instead received
      false
      at __tests__/assertion_count.test.js:14:17
  ● .assertions() › throws
    expect.assertions(2)
    Expected two assertions to be called but received one assertion call.
  ● .assertions() › throws on redeclare of assertion count
    expect(received).toBeTruthy()
    Expected value to be truthy, instead received
      false
      at __tests__/assertion_count.test.js:18:17
  ● .assertions() › throws on assertion
    expect.assertions(0)
    Expected zero assertions to be called but received one assertion call.
  ● .hasAssertions() › throws when there are not assertions
    expect.hasAssertions()
    Expected at least one assertion to be called but received none.
  .assertions()
    ✕ throws
    ✕ throws on redeclare of assertion count
    ✕ throws on assertion
  .hasAssertions()
    ✕ throws when there are not assertions
"
`;

exports[`works with node assert 1`] = `
" FAIL  __tests__/node_assertion_error.test.js
  ● assert

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      true
    Received:
      false
      
      at __tests__/node_assertion_error.test.js:16:3

  ● assert with a message

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      true
    Received:
      false
    
    Message:
      this is a message
      
      at __tests__/node_assertion_error.test.js:20:3

  ● assert.ok

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      true
    Received:
      false
      
      at __tests__/node_assertion_error.test.js:24:10

  ● assert.ok with a message

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      true
    Received:
      false
    
    Message:
      this is a message
      
      at __tests__/node_assertion_error.test.js:28:10

  ● assert.equal

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      2
    Received:
      1
      
      at __tests__/node_assertion_error.test.js:32:10

  ● assert.notEqual

    assert.notEqual(received, expected)
    
    Expected value not to be (operator: !=):
      1
    Received:
      1
    
    Difference:
    
    Compared values have no visual difference.
      
      at __tests__/node_assertion_error.test.js:36:10

  ● assert.deepEqual

    assert.deepEqual(received, expected)
    
    Expected value to deeply equal to:
      {\\"a\\": {\\"b\\": {\\"c\\": 6}}}
    Received:
      {\\"a\\": {\\"b\\": {\\"c\\": 5}}}
    
    Difference:
    
    - Expected
    + Received
    
      Object {
        \\"a\\": Object {
          \\"b\\": Object {
    -       \\"c\\": 6,
    +       \\"c\\": 5,
          },
        },
      }
      
      at __tests__/node_assertion_error.test.js:40:10

  ● assert.deepEqual with a message

    assert.deepEqual(received, expected)
    
    Expected value to deeply equal to:
      {\\"a\\": {\\"b\\": {\\"c\\": 7}}}
    Received:
      {\\"a\\": {\\"b\\": {\\"c\\": 5}}}
    
    Message:
      this is a message
    
    Difference:
    
    - Expected
    + Received
    
      Object {
        \\"a\\": Object {
          \\"b\\": Object {
    -       \\"c\\": 7,
    +       \\"c\\": 5,
          },
        },
      }
      
      at __tests__/node_assertion_error.test.js:44:10

  ● assert.notDeepEqual

    assert.notDeepEqual(received, expected)
    
    Expected value not to deeply equal to:
      {\\"a\\": 1}
    Received:
      {\\"a\\": 1}
    
    Difference:
    
    Compared values have no visual difference.
      
      at __tests__/node_assertion_error.test.js:48:10

  ● assert.strictEqual

    assert.strictEqual(received, expected)
    
    Expected value to be (operator: ===):
      NaN
    Received:
      1
      
      at __tests__/node_assertion_error.test.js:52:10

  ● assert.notStrictEqual

    assert.notStrictEqual(received, expected)
    
    Expected value not to be (operator: !==):
      1
    Received:
      1
    
    Message:
      My custom error message
    
    Difference:
    
    Compared values have no visual difference.
      
      at __tests__/node_assertion_error.test.js:56:10

  ● assert.deepStrictEqual

    assert.deepStrictEqual(received, expected)
    
    Expected value to deeply and strictly equal to:
      {\\"a\\": 2}
    Received:
      {\\"a\\": 1}
    
    Difference:
    
    - Expected
    + Received
    
      Object {
    -   \\"a\\": 2,
    +   \\"a\\": 1,
      }
      
      at __tests__/node_assertion_error.test.js:60:10

  ● assert.notDeepStrictEqual

    assert.notDeepStrictEqual(received, expected)
    
    Expected value not to deeply and strictly equal to:
      {\\"a\\": 1}
    Received:
      {\\"a\\": 1}
    
    Difference:
    
    Compared values have no visual difference.
      
      at __tests__/node_assertion_error.test.js:64:10

  ● assert.ifError

    Error
      1 thrown

  ● assert.doesNotThrow

    assert.doesNotThrow(function)
    
    Expected the function not to throw an error.
    Instead, it threw:
      [Error: err!]
    
    Message:
      Got unwanted exception.
      
      at __tests__/node_assertion_error.test.js:72:10

  ● assert.throws

    assert.throws(function)
    
    Expected the function to throw an error.
    But it didn't throw anything.
    
    Message:
      Missing expected exception.
      
      at __tests__/node_assertion_error.test.js:78:10

  ✕ assert
  ✕ assert with a message
  ✕ assert.ok
  ✕ assert.ok with a message
  ✕ assert.equal
  ✕ assert.notEqual
  ✕ assert.deepEqual
  ✕ assert.deepEqual with a message
  ✕ assert.notDeepEqual
  ✕ assert.strictEqual
  ✕ assert.notStrictEqual
  ✕ assert.deepStrictEqual
  ✕ assert.notDeepStrictEqual
  ✕ assert.ifError
  ✕ assert.doesNotThrow
  ✕ assert.throws

"
`;
